### ISSUE TITLE
 cabal new-build  problem with  GHC_PACKAGE_PATH 

### ISSUE THREAD

**Issue by andrewufrank** [leksah/leksah#418]
i installed the current 0.16.2.2 version of leksah following the instructions using the stack method (which worked nearly - see other report). the result executes but gives for the simples program (my own or leksah-welcome) the error message:

cabal new-build 'leksah-welcome:exe:leksah-welcome' 'leksah-welcome:test:test-leksah-welcome'
cabal: Use of GHC's environment variable GHC_PACKAGE_PATH is incompatible with
Cabal. Use the flag --package-db to specify a package database (it can be used
multiple times).

i do not see any way to proceed from here - leksah is not usable in this configuration (debian stretch, ghc 8.0.1, cabal 1.24.0.0). what is wrong?

**Comment by andrewufrank** [comment]
i found that this error message appears when there is no stack.yaml file present. if the package is previously intialized for stack build, this messages does not appear and the package is build with `stack build <target> --test --no-run-test` which seems ok. 
however, this opens the question, how to produce the yaml files for interdependent multi-package projects. stack deals with the solver and builds yaml files above the multiple packages - leksah still needs a cabal file to add something to the workspace. i tried to construct a fake .cabal, but did not succeed and get then an error 'stack build <target>` when no such hierarchically higher target is available. 

recommendation:  
add a method for leksah to find the stack.yaml file upwards from the current place (this is what stack build <target> does - unless there is a stack.yaml file present in the directory (which i do not see how to cosntruct when dependencies exist). if there is no stack.yaml file, then cabal new-build is used and fails.) perhaps the simplest solution is to add a switch in preferences to select either cabal or stack build processes - mixing the two in a project is not likely a good idea!

**Comment by andrewufrank** [comment]
following up from previous analysis: 
if no stack.yaml file is anywhere in the project, leksah uses cabal new-xx commands. it is then necessary to construct the `cabal.project.local` files, for which i did not find much documentation. in my case, where i have a directory with a few packages containing another directory with a few packages, the file has the following two entries: 

````
packages: /home/frank/Workspace8/jan05/repo8/uniform/*/*.cabal
packages: /home/frank/Workspace8/jan05/repo8/*/*.cabal
````

for one of the projects. test `cabal new-build` for the package, it works the same in leksah 0.16.2.2.

I hope this helps and makes it into the documentation. whether stack or cabal new-xx will be more successful is difficult to tell. installing a switch to select one or the other method may be the better way for leksah then to rely on an automatic selection based on the presence or absence of certain files. 

please document the new features !
andrew

**Comment by andrewufrank** [comment]
it works - but dangerously: if ever the 'configure' button is touched, the cabal.project.local is set back. 

given that leksah builds now with stack, i would prefer to go for a only stack based solution; i observe that cabal new-build is (at least with my project files) repeating a lot of work, which stack would not.

**Comment by andrewufrank** [comment]
the problem remains that in leksah the search for the stack.yaml file is in the directory of the cabal file and does not progress to the encompassing directories and search there for the stack.yaml (which stack build would do).
workaround: add a stack.yaml file to the directory where the cabal is. it looks like 

````
flags: {}
extra-package-dbs: []
packages:
- '.'
- ../litText
- ../rdf4store
- ../rdfhextension
- ../xx/uniform-algebras/
extra-deps:

resolver: lts-8.1
````

listing the relative path to the source of local packages   then populate the extra-deps with `stack solver`

seems to work.

**Comment by andrewufrank** [comment]
the issue with the above solution is that changes in dependent packages are not automatically integrated, because each package has its own stack.yaml. a possible solution is to construct an unnecessary cabal project in the root of the project (which is above all packages, possibly several levels deep). this `stackroot.cabal` can then be opended in leksah and uses the `stack.yaml` file there. changes in any dependent package are leading to recompilation, as expected (but effectively reconstructing too much which was not changed).  
the `stackroot.cabal` includes a trivial `main.hs` to make `stack build` succeed. this makes leksah 0.16.2.2 working with stack.

ps. thank you for the new icons! - could somebody fix  #299 - moving the panes? this is now the most annoying issue.

**Comment by andrewufrank** [comment]
solved in the current version (january 2018, 16.3.1)
